pkg/controller/storageversiongc/gc_controller.go

```diff
@@ -265,32 +266,13 @@ func (c *Controller) enqueueLease(obj *coordinationv1.Lease) {
 	c.leaseQueue.Add(obj.Name)
 }
 
-func setCommonEncodingVersion(sv *apiserverinternalv1alpha1.StorageVersion) {
-	if len(sv.Status.StorageVersions) == 0 {
-		return
-	}
-	firstVersion := sv.Status.StorageVersions[0].EncodingVersion
-	agreed := true
-	for _, ssv := range sv.Status.StorageVersions {
-		if ssv.EncodingVersion != firstVersion {
-			agreed = false
-			break
-		}
-	}
-	if agreed {
-		sv.Status.CommonEncodingVersion = &firstVersion
-	} else {
-		sv.Status.CommonEncodingVersion = nil
-	}
-}
-
 func (c *Controller) updateOrDeleteStorageVersion(sv *apiserverinternalv1alpha1.StorageVersion, serverStorageVersions []apiserverinternalv1alpha1.ServerStorageVersion) error {
 	if len(serverStorageVersions) == 0 {
 		return c.kubeclientset.InternalV1alpha1().StorageVersions().Delete(
 			context.TODO(), sv.Name, metav1.DeleteOptions{})
 	}
 	sv.Status.StorageVersions = serverStorageVersions
-	setCommonEncodingVersion(sv)
+	storageversion.SetCommonEncodingVersion(sv)
```

Comment:
is updateOrDeleteStorageVersion operating on an object obtained from an informer or a live lookup version?

Reply:
a live lookup version